addons/website_livechat/tests/common.py

```diff
@@ -18,13 +18,15 @@ def setUp(self):
             'login': 'operator',
             'email': 'operator@example.com',
             'password': "ideboulonate",
-            'livechat_username': 'El Deboulonnator',
             'groups_id': [(6, 0, [
                 self.group_user.id,
                 self.group_livechat_user.id,
             ])],
         })
-
+        self.env['res.users.settings']._find_or_create_for_user(self.operator)
+        self.operator.write({
+            'livechat_username': "El Deboulonnator",
+        })
```

Comment:
It would be best if the `livechat_username` and `livechat_lang_ids` defined their "compute/inverse" in python to automatically create the user setting if necessary, instead of having to do manually.

This is especially true as it would probably crash the form view on save if the user tries to do it but doesn't have the settings created yet.